pocket/core/core.go

```diff
@@ -177,7 +179,11 @@ func (e *Executor) Start() error {
 			go e.startBinlogTestDataCompare()
 		}
 	}
-	return nil
+
+	select {
+	case <-ctx.Done():
+		return nil
```

Comment:
It is not considered block here, return here will not stop the executation. May start a go routine to wait for ctx.Done signal and call `Stop` function.